src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -36,45 +49,46 @@ const ChooseOffer = () => {
     offerId: defaultOfferId,
   };
 
-  const determineSwitchDirection = () => {
-    const currentPeriod = subscription?.period;
+  const closeModal = useEventCallback((replace = false) => {
+    navigate(removeQueryParam(location, 'u'), { replace });
+  });
```

Comment:
Do you need dependencies on navigate and location here and below?

Reply:
It should be in a regular `useCallback`, but not in the [useEventCallback](https://github.com/reactjs/rfcs/pull/220).

We've found that when a modal has an effect, memo or callback on the location/navigate properties, all will trigger an update when closing the modal because of the location update.

Because this is hard to discover, we've chosen to wrap these dependant functions with `useEventCallback` to ensure functions depending will not update unnecessarily (and possibly cause issues).

For example, this `useEffect`:

```ts
useEffect(() => {
    // close auth modal when there are no offers defined in the config
    if (!isLoading && !offers.length) {
      closeModal(true);
    }
  }, [isLoading, offers, closeModal]);
```

This effect closes the modal (only once) based on the reactive variables `isLoading` and `offers`. The `closeModal` is still considered a reactive variable, but never updates because wrapped in `useEventCallback`.

The problem without the `useEventCallback` is that we can't ensure that this effect is only running when `isLoading` or `offers` are changed. Because it will also run when `location` changes (after a navigate).